flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartBackoffTimeStrategyFactoryLoader.java

```diff
@@ -0,0 +1,393 @@
+			setNoRestartStrategyToConfiguration(adaptedConfiguration);
+
+			return true;
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FixedDelayRestartStrategyConfiguration) {
+			final RestartStrategies.FixedDelayRestartStrategyConfiguration fixedDelayConfig =
+				(RestartStrategies.FixedDelayRestartStrategyConfiguration) restartStrategyConfiguration;
+
+			setFixedDelayRestartStrategyToConfiguration(
+				adaptedConfiguration,
+				fixedDelayConfig.getDelayBetweenAttemptsInterval().toMilliseconds(),
+				fixedDelayConfig.getRestartAttempts());
+
+			return true;
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FailureRateRestartStrategyConfiguration) {
+			final RestartStrategies.FailureRateRestartStrategyConfiguration failureRateConfig =
+				(RestartStrategies.FailureRateRestartStrategyConfiguration) restartStrategyConfiguration;
+
+			setFailureRateRestartStrategyToConfiguration(
+				adaptedConfiguration,
+				failureRateConfig.getDelayBetweenAttemptsInterval().toMilliseconds(),
+				failureRateConfig.getFailureInterval().toMilliseconds(),
+				failureRateConfig.getMaxFailureRate());
+
+			return true;
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FallbackRestartStrategyConfiguration) {
+			return false;
+		} else {
+			throw new IllegalArgumentException("Unknown restart strategy configuration " +
+				restartStrategyConfiguration + ".");
+		}
+	}
+
+	private static boolean tryAdaptingClusterRestartStrategyConfiguration(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		final String restartStrategyName = clusterConfiguration.getString(ConfigConstants.RESTART_STRATEGY, null);
+		if (restartStrategyName == null) {
+			return tryAdaptingClusterConfigurationOfDeprecatedValues(clusterConfiguration, adaptedConfiguration);
+		}
+
+		switch (restartStrategyName.toLowerCase()) {
+			case "none":
+			case "off":
+			case "disable":
+				setNoRestartStrategyToConfiguration(adaptedConfiguration);
+				return true;
+			case "fixeddelay":
+			case "fixed-delay":
+				adaptClusterConfigurationOfFixedDelayRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			case "failurerate":
+			case "failure-rate":
+				adaptClusterConfigurationOfFailureRateRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	private static boolean tryAdaptingClusterConfigurationOfDeprecatedValues(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		// support deprecated ConfigConstants values
+		final int maxAttempts = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS,
+			ConfigConstants.DEFAULT_EXECUTION_RETRIES);
+
+		final Duration heartbeatPause = parseDurationFromConfig(
```

Comment:
The behavior is a bit different here.
Previously if user has configured a valid `RESTART_STRATEGY_FIXED_DELAY_DELAY` value and an invalid `AkkaOptions.WATCH_HEARTBEAT_PAUSE` value. It would be OK.
In this diff, if the `AkkaOptions.WATCH_HEARTBEAT_PAUSE` is invalid it will always fail this process.
I think it should accepted, since a valid `AkkaOptions.WATCH_HEARTBEAT_PAUSE` is required in some other critical path.